docs/wallets.md

```diff
@@ -35,14 +35,15 @@ The process for entering shares is:
 1. The user should enter the first share. To the extent possible given screen limitations, when entering share data, data should be displayed in uppercase, with visually distinct four-character windows. The first four-character window should include the `MS1` prefix, which should be pre-filled.
 1. Once the first share is fully entered, the wallet should validate the checksum and header before accepting it.
    * The user should not be able to enter mixed-case characters. The user must be able to enter all bech32 characters as well as `?` indicating an erasure. Wallets may allow users to enter non-bech32 characters, at their discretion. (This may be useful to guide error correction, e.g. by attempting to replace all `B`s with `8`s.)
-   * If the header is invalid, the wallet should highlight this and prevent the user from entering additional data until it is fixed. An invalid header is one that starts with a character other than `0` or `2` through `9`, or one which starts with `0` but whose share index is not `S`. For shares after the first, a header is also invalid if its threshold and identifier do not match those of the first share.
-   * If the checksum is invalid, the wallet may use an error-correction algorithm to determine a corrected share, but the wallet MUST show these corrections to the user rather than silently applying them.
-   * For substitution errors, the wallet may highlight the offending 4-character window or the offending character. It may also show the corrected character.
-   * If the wallet can determine insertion or deletion errors, it should highlight the offending 4-character window. When detecting insertion or deletion errors, the wallet may assume that the correct share length is a multiple of 16 bytes.
+   * If the header is invalid, the wallet SHOULD highlight this and request confirmation from the user before allowing additional data to be entered. An invalid header is one that starts with a character other than `0` or `2` through `9`, or one which starts with `0` but whose share index is not `S`. For shares after the first, a header is also invalid if its threshold and identifier do not match those of the first share.
```

Comment:
Is it obviously an isolated error, or did they somehow produce an `S` share (or recover/store it)? It's not obvious to me at all that a `S` share is the result of the user mistyping the share index.

As for the threshold and identifier, while these are pre-filled, I have been assuming the user would have the ability to backspace over them and try to replace them.

Reply:
Well the solution is the same as reusing a previously used share index. Warn them and ask for confirmation to continue.

I suppose if it ends up being a valid codex32 secret then they are done importing regardless of threshold. And if it's not  valid then we have a strong error location hint just like repeated indices.

Reply:
> I see no benefit to being able to backspace over them. Any change will either introduce an error or produce an invalid set of shares. Why let them mess up?
>
> Or are you suggesting we let them input different identifiers and then do best of 3 error corrections on it?

I had something like that in mind. Maybe it's better to just refuse to let the user overwrite it. It does seem like it can pretty-much only make things worse. After all, their first share was accepted, with a passing checksum, with the other identifier. So if their next share has a different identifier, it either has a mistake in it (which would be corrected by forcing the identifier to be the same as the first one), or it's part of a different share set (in which case all bets are off).

Reply:
roconnor mentioned there is more error correction possible for invalid sets if given more elements than the threshold. So an "invalid set" repair utility could relax the share index, identifier, index, length reqs and threshold share limit to do deeper corrections across damaged shares.
This is also where a known address, electronically generated identifier and/or padding can help, but the tool would need to ask about its existence.

Reply:
>roconnor mentioned there is more error correction possible for invalid sets

We believe this, but we don't know any efficient algorithms for doing this. So I think any such thing is out of scope for this spec.